lang/values.md

```diff
@@ -327,7 +327,7 @@ impl<Provenance: Eq + Clone> DefinedRelation for Pointer<Provenance> {
 The order on `List<AbstractByte>` is assumed to be such that `bytes1 <= bytes2` if and only if they have the same length and are bytewise related by `<=`.
 In fact, we define this to be in general how lists are partially ordered (based on the order of their element type):
 ```rust
-impl<T: DefinedRelation> DefinedRelation for List<T> {
+impl<T: DefinedRelation + specr::Obj> DefinedRelation for List<T> {
```

Comment:
I think this will come up more and more, you're probably right that we want some automated solution for those at some point.

Reply:
Yeah, we probably just want this added to all generic parameters.

Reply:
then again we have the `FnOnce` problem. `mutate_cur_frame`s generic function argument is not `GcCompat`.

Reply:
Do we want to add some kind of opt-out, maybe?
Maybe `?specr::Obj`?

Reply:
Oh right.

Can you detect `FnOnce` and then not add the bound? (The Coq backend will also have to treat `impl FnOnce` special.)